File: app/src/main/java/com/weathercock/cleanarchitecture/domain/repository/CityRepository.java
package com.weathercock.cleanarchitecture.domain.repository;

import com.weathercock.cleanarchitecture.domain.model.city.CityModel;

import org.json.JSONException;

import java.io.IOException;
import java.util.List;

/**
 * City Repository
 *
 * @author Kensuke
 */
public interface CityRepository {

    List<CityModel> findAll() throws IOException, JSONException;

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/repository/city/CityRepository.kt
package com.kazakago.cleanarchitecture.domain.repository.city

import com.kazakago.cleanarchitecture.domain.model.city.City
import io.reactivex.Observable
import java.io.IOException

interface CityRepository {

    @Throws(IOException::class)
    fun findAll(): Observable<City>

}
